Adding a crate that exposes a binary and a library breaks corrosion

Building a crate which contains a `lib.rs` and a `main.rs` (and therefore builds a library and an executable) results in the following errors:
```text
$ cmake .. && make  # in the build subdirectory
-- Rust Toolchain: stable-x86_64-unknown-linux-gnu
-- Rust Target: x86_64-unknown-linux-gnu
-- Defaulting Cargo to build debug
-- Using Corrosion as a subdirectory
CMake Error at corrosion/cmake/CorrosionGenerator.cmake:318 (add_executable):
  add_executable cannot create imported target "rust-lib" because another
  target with the same name already exists.
Call Stack (most recent call first):
  corrosion/cmake/CorrosionGenerator.cmake:479 (_generator_add_target)
  corrosion/cmake/Corrosion.cmake:491 (_generator_add_cargo_targets)
  CMakeLists.txt:8 (corrosion_import_crate)


CMake Error at corrosion/cmake/Corrosion.cmake:337 (add_custom_target):
  add_custom_target cannot create target "cargo-build_rust-lib" because
  another target with the same name already exists.  The existing target is a
  custom target created in source directory
  "$HOME/dev/test/2022-09-19 Rust OCCT Integration".  See
  documentation for policy CMP0002 for more details.
Call Stack (most recent call first):
  corrosion/cmake/CorrosionGenerator.cmake:329 (_add_cargo_build)
  corrosion/cmake/CorrosionGenerator.cmake:479 (_generator_add_target)
  corrosion/cmake/Corrosion.cmake:491 (_generator_add_cargo_targets)
  CMakeLists.txt:8 (corrosion_import_crate)


-- Configuring incomplete, errors occurred!
See also "$HOME/dev/test/2022-09-19 Rust OCCT Integration/build/CMakeFiles/CMakeOutput.log".

```

The crate was called `rust-lib` in this example.

Steps to reproduce:
1. create a rust crate with a library and a binary (e.g. by adding a main.rs to a lib crate)
2. add it to a `CMakeLists.txt`, e.g. the crate bing in the `rust/` subdirectory
    ```cmake
    add_subdirectory(corrosion)
    corrosion_import_crate(MANIFEST_PATH rust/Cargo.toml)
    ```
3. try to build it

This happens on the latest release (e1144c4b7aa44004a8b8c6cd254c63420ce3e309) and on the current master (c6b793604d5cdb8e688cc2f627f034865a50327d).

--- a/cmake/CorrosionGenerator.cmake
+++ b/cmake/CorrosionGenerator.cmake
@@ -55,6 +55,21 @@ function(_generator_add_package_targets workspace_manifest_path package_manifest
             list(APPEND kinds ${kind})
         endforeach()
 
+        if(TARGET "${target_name}"
+            AND ("staticlib" IN_LIST kinds OR "cdylib" IN_LIST kinds OR "bin" IN_LIST kinds)
+            )
+            message(WARNING "Failed to import Rust crate ${target_name} (kind: `${target_kind}`) because a target "
+                "with the same name already exists. Skipping this target.\n"
+                "Help: If you are importing a package which exposes both a `lib` and "
+                "a `bin` target, please consider explicitly naming the targets in your `Cargo.toml` manifest.\n"
+                "Note: If you have multiple different packages which have targets with the same name, please note that "
+                "this is currently not supported by Corrosion. Feel free to open an issue on Github to request "
+                "supporting this scenario."
+                )
+            # Skip this target to prevent a hard error.
+            continue()
+        endif()
+
         if("staticlib" IN_LIST kinds OR "cdylib" IN_LIST kinds)
             if("staticlib" IN_LIST kinds)
                 set(has_staticlib TRUE)

